src/lib/blob_route.hpp

```diff
@@ -12,12 +12,12 @@ namespace homeobject {
 // to appear in a different Index should the Blob (Shard) be moved between Pgs.
 #pragma pack(1)
 struct BlobRoute {
+    enum KeyType { kTypeTombStone = 0x0, kTypeNormal = 0x1 };
+    KeyType keyType;
```

Comment:
I will suggest to move keyType to behind of blob and use shard_id as the highest bits, as we need to range query all blobs for one shard.

Reply:
Should the Route be where we store the tombstone and not an extension of the Index value (as the memory_backend is doing it)?

Reply:
actually, we want to do a rangequery  to find out all the tomestone blob in a given shard. 
in a common KV DB(for example, leveldb), we should provide the key prefix, which should be `shardID+tombstone` in our use case to get the best efficiency , just like @zichanglai said.

so , i prefer to in put the tombstone after shardid in Route for now, since this will get a better efficiency than an extension of the Index value

Reply:
How do we know all blobs are deleted as each is independen operation to create a shardID + tombstone key ? Having a key prefix for a blob or shard might help to find if its deleted or not but dont we need to check it for put to know if its deleted or not ? Couple of options
a. Use prefix key tombstone + shard id + blob id in same tree. Cons get needs to check two keys actual and tombstone key.
b. Use a different tree to store all deleted keys (1 active and 1 pending) - Update the value in current index btree to mark it as deleted and insert into active delete tree. GC only needs to look at pending delete tree which is very small compared to the actual index btree. double buffering of active and pending tree.